compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/service.bal

```diff
@@ -111,4 +113,20 @@ service http:Service on new http:Listener(9090) {
         AA val = { a: "hello" };
         error? a = abc->respond(val); // error
     }
+
+    resource function get callerInfo19(@http:CallerInfo {respondType: Person} http:Caller abc) returns error? {
+        check abc->respond(error("hello world")); // error
+    }
+
+    resource function get callerInfo20(@http:CallerInfo {respondType: http:Error} http:Caller abc) returns error? {
+        check abc->respond(<http:Error> error("hello world"));
+    }
+
+    resource function get callerInfo21(@http:CallerInfo {respondType: http:Error} http:Caller abc) returns error? {
+        check abc->respond(error("hello world")); // error
+    }
+
+    resource function get greeting(@http:CallerInfo{respondType: NewError} http:Caller caller) returns error? {
```

Comment:
Add a test with `{respondType: error}` as negative case. So that once this is allowed from lang team, we can turn it to a success test

Reply:
Added in [67efe99](https://github.com/ballerina-platform/module-ballerina-http/pull/960/commits/67efe99cd1f36009b67c5cb2e4895c341f0bf0f5)